test/core/rootfind_tests.jl

```diff
@@ -55,7 +55,7 @@ end
 @testitem "NewtonRaphson" setup=[CoreRootfindTesting] tags=[:core] timeout=3600 begin
     @testset "LineSearch: $(_nameof(lsmethod)) LineSearch AD: $(_nameof(ad))" for lsmethod in (
             Static(), StrongWolfe(), BackTracking(), HagerZhang(), MoreThuente()),
-        ad in (AutoFiniteDiff(), AutoZygote())
```

Comment:
we should probably keep testing `AutoFiniteDiff`

Reply:
+1 Keep FiniteDiff, just add ForwardDiff

Reply:
@avik-pal  If using FiniteDiff, should we consider using it on $\phi(\alpha)$, instead of on the gradient? i.e.,

$$
\phi'(\alpha_0)=\frac{\phi(\alpha_0+\varepsilon)-\phi(\alpha_0-\varepsilon)}{2}
$$

This should be as accurate and much faster than FiniteDiff-ing the gradient

Reply:
yeah makes sense